### ISSUE TITLE
docs: link to typescript page is dead in Angular Query Options

### ISSUE THREAD

**Issue by ledenis** [TanStack/query#8022]
### Describe the bug

In [Query Options page for Angular](https://tanstack.com/query/latest/docs/framework/angular/guides/query-options), the link ["with TypeScript"](https://tanstack.com/query/latest/docs/framework/angular/typescript#typing-query-options) is a 404.

### Your minimal, reproducible example

N/A

### Steps to reproduce

1. Go to https://tanstack.com/query/latest/docs/framework/angular/guides/query-options
2. Click on ["with TypeScript"](https://tanstack.com/query/latest/docs/framework/angular/typescript#typing-query-options) link
3. Observe a 404 error

### Expected behavior

Should navigate to a page that shows the equivalent of this content for Angular: https://github.com/TanStack/query/blob/main/docs/framework/react/typescript.md#typing-query-options

### How often does this bug happen?

Every time

### Platform

N/A

### Tanstack Query adapter

angular-query

### TanStack Query version

N/A

**Comment by DogPawHat** [comment]
Quick fix is to remove the broken link by placing a `[//]: # 'Description'` directive (or whatever the yoke is called) around the text block in `react/guides/query-options.md` and add replacement text in `angular/guides/query-options.md` without the link.

The proper fix is to write up `angular/typescript.md` properly. It should share a lot of content with the React and Vue versions, but I've not touched Angular in a hot minute.

**Comment by arnoud-dv** [comment]
Yes it needs to be written. It would be different from React and Vue as signals being functions and typescript currently not supporting discriminated unions on properties returning a function brings some typing challenges.

**Comment by arnoud-dv** [comment]
I just added the TypeScript page: https://tanstack.com/query/latest/docs/framework/angular/typescript
